plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerClient.java

```diff
@@ -369,7 +372,7 @@ public String getBindExpression()
             {
                 // Binding setObject(LocalTime) can result with "The data types time and datetime are incompatible in the equal to operator."
                 // when write function is used for predicate pushdown.
-                return format("CAST(? AS time(%s))", precision);
+                return format("CAST(? AS time(%s))", min(precision, MAX_SUPPORTED_TEMPORAL_PRECISION));
```

Comment:
isn't precision known not to exceed MAX_SUPPORTED_TEMPORAL_PRECISION at this point?

Reply:
No, there can be a literal value with precision up to 12.

Reply:
i though you `min` with `MAX_SUPPORTED_TEMPORAL_PRECISION` before calling the write function

Reply:
You are right, removed!